### PR TITLE
Add action for extra field in admin search form

### PR DESCRIPTION
**Pull request by abhishek-kaushik**
We wanted to add a new search field in `Zone Content` to make search more refined.
But currently `zoninator` plugin does not contain any hooks to achieve this functionality.
So I have added a new action `zoninator_admin_search_form_content` just to hook new search content.

### CODE REVIEW

File: zoninator.php [hunk 1]
```diff
@@ -448,6 +448,8 @@ function admin_page_zone_edit( $zone = null ) {
 							<?php $this->zone_admin_recent_posts_dropdown( $zone_id ); ?>
 
 							<?php $this->zone_admin_search_form(); ?>
+							
+							<?php do_action( 'zoninator_admin_search_form_content' ); ?>
```
Comment (dhanendrannewscorp) [inline zoninator.php]: @abhishek-kaushik I think it's good to pass the `$zone_id` to the action. Also can you document this new `action`?

**Comment by mjangda** [comment]
Can you please provide an example of how this would be used?

**Comment by kaushik-news-co-uk** [comment]
We wanted to get the post by post ID. So Added new textbox to put the post id and then fetch the post to add in zone

**Comment by mjangda** [comment]
My suggestion would be to use the existing search input field to allow this. You can hook in to `zoninator_search_args` and change the search query as needed.

```
add_filter( 'zoninator_search_args', function( $args ) {
    if ( is_numeric( $_REQUEST['term'] ) ) {
        $args = array(
            'post__in' => array(
                absint( $_REQUEST['term'],
            ),
            'no_found_rows' => true,
        );
    }

    return $args;
} );
```

**Comment by abhishek-kaushik** [comment]
@mjangda , We have pitched this solution to our editorial team.
But they want a different text box for article IDs.

**Comment by philipjohn** [comment]
Noting that we chatted privately about this and @mjangda's suggested will work for the requirements in this case.

**Pull request closed without merging**